funfact/algorithm.py

```diff
@@ -65,7 +73,11 @@ def factorize(
             that represents all the solutions.
     '''
 
-    target = ab.tensor(target)
+    if dtype == 'target':
+        dtype = str(target.dtype)
+        dtype = operator.attrgetter(dtype)(ab)
```

Comment:
I suppose this is, for example, to convert numpy dtypes to the equivalent ones in the backend?

How about flipping the order of this `if` block with line 80, so that we can directly use the type of the new `target` tensor?

Reply:
The idea by first converting it to a native type tensor was that `dtype = str(target.dtype)` always gives a dtype from the backend. 

For example with torch this will become something like `'torch.float32'` which can be found by `operator.attrgetter`. If we flip it around, we can get something like `'numpy.float3'` which would cause an error in line 78 as it is not an attr of `ab`.

Reply:
Oh, I meant what if we do this:
``` py
if dtype is None:
    target = ab.tensor(target)
    dtype = target.dtype
else:
    target = ab.tensor(target, dtype=dtype)
    dtype = target.dtype
```
and then be done. If the user wanna explicit control of dtype, they will manually specify a dtype from `ab`, or convert target to `ab.native_t` before passing it into vectorize.